### ISSUE TITLE
combined_loader: concatenating inputs

### ISSUE THREAD

**Issue by heavenlycable-three** [Lightning-AI/lightning#17804]
### Description & Motivation

Application: using the combined data loader with two datasets of the same format
Suggestion: option for  concatenating the data + labels in the combined loader
Motivation: currently one has to edit model code when data source is changed -> very unelegant solution and likely leads to errors! 
Comment: Maybe I just did not understand the documentation right. if so please improve the documentation. :-)

currently I use:
  
  ```
iterables = {"dataset1": dataloader1, "dataset2": dataloader2}
  combined_loader = CombinedLoader(iterables, mode="min_size")
  
```
  and have to unpack the data in the models training step using:
  
```
  def training_step(self,batch,batch_idx):
          #batch1,batch2 = batch
          x = torch.concat([batch['dataset1'][0],batch['dataset2'][0]])
          y = torch.concat([batch['dataset1'][1],batch['dataset2'][1]])

```

### Pitch

It would be great to have something like this (example for regression tasks on images):
```
dataloader1 = DataLoader(..., batch_size = b1)
dataloader2 = DataLoader(..., batch_size = b2)

iterables = {"dataset1": dataloader1, "dataset2": dataloader2}
combined_loader = CombinedLoader(iterables, mode = 'min_size', operation = 'concat')
x,y = next(iter(combined_loader)
# x.shape = [b1+b2,h,w,c]
# y.shape = [b1+b2]

```
